VIC_RBM/model_run/result_analysis/scripts/my_functions.py

```diff
@@ -527,5 +544,64 @@ def plot_seasonality_data(list_s_data, list_style, list_label, plot_start, plot_
 
 	return fig
 
+#==============================================================
+#==============================================================
+
+def plot_WY_mean_data(list_s_data, list_style, list_label, plot_start, plot_end, xlabel=None, ylabel=None, title=None, fontsize=16, legend_loc='lower right', time_locator=None, time_format='%Y/%m', add_info_text=False, model_info=None, stats=None, show=False):
+	''' This function plots annual mean (water year) data time series
+
+	Require:
+		plot_date_format
+		add_info_text_to_plot(fig, ax, model_info, stats)
+		plot_time_series
+	'''
+
+	# Check if list_s_data, list_style and list_label have the same length
+	if len(list_s_data) !=len(list_style) or len(list_s_data)!=len(list_label):
+		print 'Input list lengths are not the same!'
+		exit()
+
+	# Calculate WY mean data
+	list_s_WY = []   # list of new monthly mean data in pd.Series type 
+	for i in range(len(list_s_data)):
+		s_month = calc_monthly_data(list_s_data[i])
+		list_s_month.append(s_month)
+
+	# plot
+	fig = plot_time_series(True, list_s_month, list_style, list_label, plot_start, plot_end, xlabel, ylabel, title, fontsize, legend_loc, time_locator, time_format, add_info_text=add_info_text, model_info=model_info, stats=stats, show=show)
+
+	return fig
+
+#==============================================================
+#==============================================================
+
+def plot_WY_mean_data(list_s_data, list_style, list_label, plot_start, plot_end, xlabel=None, ylabel=None, title=None, fontsize=16, legend_loc='lower right', time_locator=None, time_format='%Y/%m', add_info_text=False, model_info=None, stats=None, show=False):
+	''' This function plots water year annual mean data time series
+
+	Require:
+		plot_date_format
+		add_info_text_to_plot(fig, ax, model_info, stats)
+		plot_time_series
+		calc_ts_stats_by_group
+	'''
+
+	# Check if list_s_data, list_style and list_label have the same length
+	if len(list_s_data) !=len(list_style) or len(list_s_data)!=len(list_label):
+		print 'Input list lengths are not the same!'
```

Comment:
Also, this is a test that would be useful in any of the plotting functions you have, so you could either pull it out to happen at the beginning of the script entirely or you could collapse all of the plotting function such that the test is always read when plotting (no matter on what time frame).